jwst/extract_1d/extract.py

```diff
@@ -2562,14 +2563,25 @@ def run_extract1d(input_model, refname, smoothing_length, bkg_order,
         obtained by iterating over a SourceModelContainer.  The default
         is False.
 
+    apcorr_ref_name : str
+        Name of the APCORR reference file. Default is None
+
     Returns
     -------
     output_model : data model
         A new MultiSpecModel containing the extracted spectra.
-    """
 
+    """
     # Read and interpret the reference file.
-    ref_dict = load_ref_file(refname)
+    ref_dict = load_ref_file(extract_ref_name)
+
+    apcorr_table = apcorr_sizeunits = None
+    if apcorr_ref_name is not None or apcorr_ref_name != 'N/A':
+        with fits.open(apcorr_ref_name) as ap_ref:
+            apcorr_table = ap_ref['APCORR'].data.copy()
+            apcorr_sizeunits = ap_ref['APCORR'].header['SIZEUNIT']
```

Comment:
If we loaded apcorr into a data model, you could then just pass the data model around, rather than having to store and pass around separate objects containing the table data, units, etc.